src/Network/Google/Drive/File.hs

```diff
@@ -133,12 +134,12 @@ getFile fid = (Just <$> getJSON (fileUrl fid) [])
 -- | Create a @File@ from @FileData@
 createFile :: FileData -> Api File
 createFile fd =
-    postJSON (baseUrl <> "/files") [("setModifiedDate", Just "true")] fd
+    postJSON (baseUrl <> "/files") (maybe [] (const [("setModifiedDate", Just "true")]) (fileModified fd)) fd
```

Comment:
I think the use of `maybe`/`const` (which I usually prefer) obscures what's going on this case. I wonder if an explicit pattern match would make things clearer:

```hs
postJSON (...) (params $ fileModified fd) fd
  where
    params (Just _) = [("setModifiedDate", Just "true")]
    params Nothing = []
```

What do you think?

Same comment for the `putJSON` call below.